channels/urbdrc/client/libusb/libusb_udevman.c

```diff
@@ -667,56 +598,62 @@ static void udevman_load_interface(UDEVMAN* udevman)
 	udevman->iface.initialize = udevman_initialize;
 }
 
-static BOOL urbdrc_udevman_register_devices(UDEVMAN* udevman, const char* devices)
+static BOOL udevman_parse_device_id_addr(char** str, UINT16* id1, UINT16* id2, UINT16 max,
+                                         char split_sign, char delimiter)
 {
-	BOOL rc = FALSE;
-	char* token;
-	int success = 0;
-	char* tmp;
-	char hardware_id[16];
-	const char* default_devices = "id";
-
-	if (!devices)
-		tmp = _strdup(default_devices);
-	else
-		tmp = _strdup(devices);
+	char* mid;
+	unsigned long rc;
+
+	rc = strtoul(*str, &mid, 16);
+
+	if ((mid == *str) || (*mid != split_sign) || (rc > max))
+		return FALSE;
+
+	*id1 = (UINT16)rc;
+	rc = strtoul(++mid, str, 16);
+
+	if ((*str == mid) || (rc > max))
+		return FALSE;
 
-	/* register all usb devices */
-	token = strtok(tmp, "#");
+	*id2 = (UINT16)rc;
 
-	while (token)
+	if (**str == '\0')
+		return TRUE;
+	if (**str == delimiter)
 	{
-		strcpy(hardware_id, token);
-		token = strtok(NULL, "#");
+		(*str)++;
+		return TRUE;
+	}
+
+	return FALSE;
+}
 
+static BOOL urbdrc_udevman_register_devices(UDEVMAN* udevman, char* devices)
```

Comment:
It might be a nice addition to make this `const char* device` to indicate that we don´t modify the content.